Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Projection/ProjectionIntegrityViolationDetector.php

```diff
@@ -67,8 +67,8 @@ public function hierarchyIntegrityIsProvided(): Result
         }
 
         $invalidlyHashedHierarchyRelationRecords = $this->client->getConnection()->executeQuery(
-            'SELECT * FROM ' . $this->tableNamePrefix . '_hierarchyrelation
-                WHERE dimensionspacepointhash != MD5(dimensionspacepoint)'
+            'SELECT * FROM ' . $this->tableNamePrefix . '_hierarchyrelation h LEFT JOIN ' . $this->tableNamePrefix . '_dimensionspacepoints dsp ON dsp.hash = h.dimensionspacepointhash
```

Comment:
I don't know what the purpose for this check is. But after it's not fully the same as before. So we just check if the entry for a given hash is present in `_dimensionspacepoints`. But before we also checked if the md5-hashed value is equal to the hash.

Reply:
Mmmm, we can't know if the hash is the right one though, either it exists in the DSP table or it's invalid, if it exists in there IMHO we can assume it matches the dimension coordinates. We _could_ add a separate check that ready all DSP entries and compares hash and getting the hash from the coordinates in the same row if we really want to.

Reply:
I'm fine with keeping it as it is. Just wanted to point out this difference.